crates/plugin-api/src/wasi_p1.rs

```diff
@@ -44,32 +44,36 @@ unsafe extern "C" fn cabi_realloc(
     let new_mem = match new_size {
         0 => ZERO_SIZE_ALLOCATION_PTR,
         // this call to `alloc` is safe since `new_size` must be > 0
-        _ => alloc::alloc(Layout::from_size_align(new_size, alignment).unwrap()),
+        _ => unsafe { alloc::alloc(Layout::from_size_align(new_size, alignment).unwrap()) },
```

Comment:
Should we remove `unsafe`s on the functions if we're adding it in the implementation?